### ISSUE TITLE
Evaluate "contrastive loss"

### ISSUE THREAD

**Issue by hominot** [hominot/research#19]
- [ ] Decide on batch size.
  - K80 12GB gpu can run batch size of 32. Batch size of 48 gives OOM.
  - Try batch size of 64, 96, 128 etc on M40 24GB.
  - Let's use the same batch size for all experiments.
- [ ] Tune hyperparameters:
  - Learning rate: trainer.jinja2 learning_rate
  - Batch design: dataset.jinja2 group_size, num_groups (usually, num_groups = batch_size / group_size)
  - For tuning, perform a grid search. For each grid point, run one or two experiments and compare validation error to decide on the best hyperparameter setting.
  - Example configs: https://github.com/hominot/research/blob/master/metric_learning/experiments/contrastive_tuning.json
- [ ] Run cross validation on test dataset
  - Example script: python run.py --split=0/1/2/3/4
  - Examplt script: python evaluate.py --experiment={{ run_name }} --step={{ checkpoint_number }}
  - report average/stdev metrics
